File: OboeTemplate/app/src/main/java/com/gmail/meeyeer/viinceent/oboetemplate/OpenGLTouchActivity.java
package com.gmail.meeyeer.viinceent.oboetemplate;

import android.content.res.AssetManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.method.ScrollingMovementMethod;
import android.view.WindowManager;
import android.widget.TextView;

public class OpenGLTouchActivity extends AppCompatActivity {

    private native void native_createOboeTouchClap(AssetManager assetManager);
    private native void native_startOboeSound();
    private native void native_stopOboeSound();
    private native void native_closeOboeSound();
    private native boolean native_isOboeSoundExclusive();

    private TextView statusText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_opengl_touch);
        statusText = findViewById(R.id.statusText);
        statusText.setMovementMethod(new ScrollingMovementMethod());
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        native_createOboeTouchClap(getAssets());
    }

    @Override
    protected void onResume() {
        super.onResume();
        native_startOboeSound();
        boolean isExclusive = native_isOboeSoundExclusive();
        statusText.setText(isExclusive ? "Exclusive mode" : "Shared mode");
    }

    @Override
    protected void onPause() {
        super.onPause();
        native_stopOboeSound();
    }

    @Override
    protected void onStop() {
        super.onStop();
        native_closeOboeSound();
    }
}

File: OboeTemplate/app/src/main/kotlin/com/gmail/meeyeer/viinceent/oboetemplate/OpenGLTouchActivity.kt
package com.gmail.meeyeer.viinceent.oboetemplate

import android.content.res.AssetManager
import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.gmail.meeyeer.viinceent.oboetemplate.databinding.ActivityOpenglTouchBinding

class OpenGLTouchActivity :
    AppCompatActivity() {
    private lateinit var binding: ActivityOpenglTouchBinding
    private external fun native_createOboeTouchClap(assetManager: AssetManager)
    private external fun native_startOboeSound()
    private external fun native_stopOboeSound()
    private external fun native_closeOboeSound()
    private external fun native_isOboeSoundExclusive(): Boolean

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityOpenglTouchBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.statusText.movementMethod = ScrollingMovementMethod()
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        native_createOboeTouchClap(assets)
    }

    override fun onResume() {
        super.onResume()
        native_startOboeSound()
        val isExclusive = native_isOboeSoundExclusive()
        binding.statusText.text = if (isExclusive) "Exclusive mode" else "Shared mode"
    }

    override fun onPause() {
        super.onPause()
        native_stopOboeSound()
    }

    override fun onStop() {
        super.onStop()
        native_closeOboeSound()
    }
}
